tests/processor/document/test_split_into_partition.py

```diff
@@ -0,0 +1,126 @@
+import pytest
+
+from pie_utils.processor.document import DocumentWithPartition
+from pie_utils.processor.document.split_into_partition import SplitDocumentToPartitions
+
+TEXT1 = """<start>Jane lives in Berlin. this is no sentence about Karl.
+<middle>Seattle is a rainy city. Jenny Durkan is the city's mayor.
+<end>Karl enjoys sunny days in Berlin.
+"""
+
+TEXT2 = """<start>Lily is mother of Harry.
+<end>Beth greets Emma.
+"""
+
+
+def test_split_document_to_partitions():
+    split_document_to_partitions = SplitDocumentToPartitions(
+        pattern="(<start>|<middle>|<end>)",
+        label_group_id=0,
+        label_whitelist=["<start>", "<middle>", "<end>"],
```

Comment:
Can you test for the actual partition labels?

Reply:
Do you mean using H1, Title etc as label ? Also, in all examples or just this one ?

Reply:
no, just check that the `partitions[0].label == "<start>"` etc

Reply:
and one other thing: can you not include all labels in `label_whitelist`? so that we really test that non-whitelist labels are not used.

Reply:
fixed in [18bbfe8](https://github.com/ArneBinder/pie-utils/pull/21/commits/18bbfe8640dbbaa8d8108162836663e133dc5d11)

Reply:
All cases for `label_whitelist` is tested with [250f47f](https://github.com/ArneBinder/pie-utils/pull/21/commits/250f47f7c9f8b97f4541d39e312f7654a722c2ee) commit. There is a case if we pass empty list as `label_whitelist` which will result in no partitions except for maybe initial_part if that is not skipped. Should we test this case as well ?

Reply:
good catch, yes, let's add the case where no parts are added. This can also happen, if the matcher does not return any match.

Reply:
Test cases added in [b461ce0](https://github.com/ArneBinder/pie-utils/pull/21/commits/b461ce0d6b85d1a4308ba630ff361bfc5b6d290d)